include translation documentation (#1436)
btcpayserver/btcpayserver-doc: docs/Translation.md

---
description: BTCPay Translation features.
---

# Using BTCPay Translation Feature to Localize Your BTCPay Server instance

> BTCPay Server now includes a powerful translation feature that allows merchant administrators manage their instance in their preferred language. 

Reviewer: ```suggestion
Since version 2.0 BTCPay Server includes a translation feature that allows administrators to set up a default language for users accessing their instance.
```

> This guide will help you create and manage translations, making the BTCPay platform more user-friendly for you.

Reviewer: Don't think info here is adequate for a tech doc, if we use ChatGPT let's try not to make it sound like a marketer. 
```suggestion
This guide will help you create and manage translations, making the BTCPay platform more user-friendly for you.
```

> With this feature, you can replace the default English text across the entire application with your chosen language. 

Reviewer: ```suggestion
With this feature, you can replace the default English text across the entire back-office with your chosen language. 
```

> Whether you want to customize texts on your checkout page, settings configuration, dashboard, or even error and success messages, BTCPay's translation feature makes it easy to tailor your experience to your linguistic needs.

Reviewer: Not really useful information, especially it's incorrect, translations currently don't affect checkout. Only the back-office interface.
```suggestion
```

> Note: You must be a server admin to access this feature. If you're not an administrator, please introduce this functionality to your admin.

Reviewer: Can you proofread this one, I don't understand the second part of the sentence.

> ## Step-by-Step Guide to Translating Text in BTCPay Server

Reviewer: ```suggestion
## Translating BTCPay Server
```

1. Log in to your BTCPay Server instance.
2. Navigate to Translation on **Server Settings** >> **Translations**.

![Translation 1](./img/Translations/01_Translation.png)

3. Click **Create** button to generate a new language dictionary.

> 4. Enter the name you want to use to save the translation dictionary and then click **Create** button to create the dictionary. 

Reviewer: Enter the name you want to use **to save the translation dictionary**  

Not sure what this means?

As an example lets start by translating "Add Role" into Yoruba. Since I created a dictionary for Yoruba language I would translate to Yoruba language.

![Translation 5](./img/Translations/05_Translation_Add_Role_To_Yoruba.png)

Replace the text and click the **Save** button. A confirmation message will appear indicating that your translations have been successfully saved. Now click on the **Select** button for the newly created dictionary to update the default dictionary on the system. 

![Translation 6](./img/Translations/06_Translation_Saved_Dictionary.png)

Now to test out our translation. You can navigate to **Roles** under **Server Settings**, and if we look at the button on the top right of the Roles view, we can see that the text has been successfully translated from "Add Roles" to "Fi ipa kun".

![Translation 7](./img/Translations/07_Translation_Validation.png)

We can search other part of the BTCPay application that has "Add Role" text, and we can also confirm that it has successfully been translated to our chosen language text. 

> Now that you have successfully translated one, go ahead and translate other texts in the dictionary. All instances of English text will be replaced with your translated text, and now you can enjoy your new localized experience.

Reviewer: ```suggestion
Now that you have successfully translated one, go ahead and translate other texts in the dictionary. All instances of English text will be replaced with your translated text, and now you can enjoy your new localized experience.
:::tip
All translation strings are in a single text box, easy to copy over. Feel free to leverage this by pasting them into translation AI tools like ChatGPT, giving yourself a good starting point. We strongly recommend that you manually review all strings afterwards, ensuring contextual accuracy, which is sometimes lost when depending on AI.
:::
```

## Tips for Effective Translations

> - **Consistency is Key:** Ensure that similar terms are translated consistently throughout.

Reviewer: ```suggestion
- **Consistency is Key:** Ensure that similar terms are translated consistently throughout.
```

> - **Context Matters:** Keep context in mind when translating phrases to maintain meaning.

Reviewer: ```suggestion
- **Context matters:** Keep context in mind when translating phrases to maintain meaning.
```

> - **Review Regularly:** Regularly update translations as new features are added.

Reviewer: ```suggestion
- **Review regularly:** Regularly update translations as new features are added.
```

> 
> 
> The translation feature on BTCPay Server allows you to create a personalized experience by managing content in your desired language. By following the steps highlighted above, you can easily set up and maintain translations, enhancing usability for yourself and your members.
> 
> ## Support
> 
> Feel free to join our support channel over at [Mattermost chat](http://chat.btcpayserver.org/) if you need help or have any further questions.
> 
> If experience a bug or you would like to report any missing translation string, please open an issue in our [Github repository](https://github.com/btcpayserver/btcpayserver/issues)
> 
> 

Reviewer: Don't know these are just words, obvious chatgpt not really helpful so I'd remove them.
```suggestion
```
